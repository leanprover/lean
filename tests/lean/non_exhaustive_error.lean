
def g : ℕ × ℕ → ℕ
| (0, n) := n
| (m+2, 0) := m

definition f : string → nat → bool
| "hello world" 1 := tt
| "bye"         _ := tt

def foo (y : ℕ) : ℕ → ℕ → ℕ
| 0     a     := a
| (x+1) 0     := 3 + foo x 10
| (x+1) (n+2) := 42

namespace other

mutual def f, g
with f : nat → nat → nat
| 0 0 := 1
with g : nat → nat → nat
| 0 0 := 1

end other

inductive term | mk : ℕ → list term → term

def const_name : term → ℕ
| (term.mk n []) := n