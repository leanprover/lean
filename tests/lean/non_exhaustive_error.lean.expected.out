non_exhaustive_error.lean:2:4: error: non-exhaustive match, the following cases are missing:
g (1, _)
g (nat.succ (nat.succ _), nat.succ _)
non_exhaustive_error.lean:2:0: warning: declaration 'g' uses sorry
non_exhaustive_error.lean:6:11: error: non-exhaustive match, the following cases are missing:
f "hello world" 0
f "hello world" (nat.succ (nat.succ _))
f _ _
non_exhaustive_error.lean:6:0: warning: declaration 'f' uses sorry
non_exhaustive_error.lean:10:4: error: non-exhaustive match, the following cases are missing:
foo (nat.succ _) 1
non_exhaustive_error.lean:10:0: warning: declaration 'foo' uses sorry
non_exhaustive_error.lean:17:11: error: non-exhaustive match, the following cases are missing:
f 0 (nat.succ _)
f (nat.succ _) _
g 0 (nat.succ _)
g (nat.succ _) _
non_exhaustive_error.lean:17:0: warning: declaration '_mutual.f.g' uses sorry
non_exhaustive_error.lean:27:4: error: non-exhaustive match, the following cases are missing:
const_name (term.mk _ (term.mk _ _ :: _))
non_exhaustive_error.lean:27:0: warning: declaration 'const_name' uses sorry
